src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1401,12 +1402,29 @@ void LogCommand()
                 var p = s.InputParameters;
                 if (p.Count > 0 && (NpgsqlLogManager.IsParameterLoggingEnabled || connector.Settings.LogParameters))
                 {
-                    sb.Append('\t').Append("Parameters:");
                     for (var i = 0; i < p.Count; i++)
-                        sb.Append("\t$").Append(i + 1).Append(": ").Append(Convert.ToString(p[i].Value, CultureInfo.InvariantCulture));
+                    {
+                        sb.Append("\t").Append("Parameters $").Append(i + 1).Append(":");
+                        switch (p[i].Value)
+                        {
+                        case IList list:
+                            for (var j = 0; j < list.Count; j++)
+                            {
+                                sb.Append("\t#").Append(j).Append(": ").Append(Convert.ToString(list[j], CultureInfo.InvariantCulture));
```

Comment:
Is the intention here really to output all list items in the same, single row, with tabs separating them? Them seems like it would become unreadable very quickly. Should we just have a line per list item, with an additional tab beyond the parameter's?

Reply:
Yes, we have a line per list item, with each item in the list being separated by a tab and the index of the item in the list.

Example output would be: 

```
Parameters $1:  200
Parameters $2:  #0: 1   #1: 2   #2: 3
Parameters $3:  #0: aa  #1: bb  #2: CC  #3: dd
```

Reply:
Note that at the end of the for loop there is a sb.AppendLine()

Reply:
Ok. This may work with a few short items, but once you start having a few more - or just one item that's slightly larger than the tab size - it's likely to become a big mess...

Am ok to merge either way - we can always change later based on user feedback. Let me know.

Reply:
It doesn't look that way in the last version of the code - you may want to check it out. I'd I'm wrong (am on mobile) feel free to ignore.

Reply:
Am okay with any version since parameter logging is only for debugging, so the should be short arrays. The problem can occur only in case when there is an array of strings.

Reply:
It's true this feature is only for debugging, but I'm not sure why the arrays would be short... Am going to merge this anyway - let's see if we get complaints.